packages/sw-cli/src/cli/index.js

```diff
@@ -230,6 +227,91 @@ class SWCli {
     });
   }
 
+  _getFileExtensionsToCache(rootDirectory) {
+    return this._getFileContents(rootDirectory)
+    .then((files) => {
+      return this._getFileExtensions(files);
+    })
+    .then((fileExtensions) => {
+      if (fileExtensions.length === 0) {
+        throw new Error(errors['no-file-extensions-found']);
+      }
+
+      return inquirer.prompt([
+        {
+          name: 'cacheExtensions',
+          message: 'Which file types would you like to cache?',
+          type: 'checkbox',
+          choices: fileExtensions,
+          default: fileExtensions,
+        },
+      ]);
+    })
+    .then((results) => {
+      if (results.cacheExtensions.length === 0) {
+        throw new Error(errors['no-file-extensions-selected']);
+      }
+
+      return results.cacheExtensions;
+    })
+    .catch((err) => {
+      logHelper.error(
+        errors['unable-to-get-file-extensions'],
+        err
+      );
+      throw err;
+    });
+  }
+
+  _getFileContents(directory) {
+    return new Promise((resolve, reject) => {
+      fs.readdir(directory, (err, directoryContents) => {
+        if (err) {
+          return reject(err);
+        }
+
+        resolve(directoryContents);
+      });
+    })
+    .then((directoryContents) => {
+      const promises = directoryContents.map((directoryContent) => {
+        const fullPath = path.join(directory, directoryContent);
+        if (fs.statSync(fullPath).isDirectory()) {
+          if (!constants.blacklistDirectoryNames.includes(directoryContent)) {
+            return this._getFileContents(fullPath);
+          } else {
+            return [];
+          }
+        } else {
+          return Promise.resolve(fullPath);
+        }
+      });
+
+      return Promise.all(promises);
+    })
+    .then((fileResults) => {
+      return fileResults.reduce((collapsedFiles, fileResult) => {
+        return collapsedFiles.concat(fileResult);
+      }, []);
+    });
+  }
+
+  _getFileExtensions(files) {
+    const fileExtensions = {};
```

Comment:
Ah, gotcha. Then feel free to go ahead with the iterating-over-each-item approach, though you might still benefit from using a `Set` instead of an `Object` with dummy `true` values.

It's not a big deal either way.